src/Sentry/Platforms/iOS/Extensions/UserExtensions.cs

```diff
@@ -0,0 +1,72 @@
+using System.Collections.ObjectModel;
+using System.Text.Json;
+
+namespace Sentry.iOS.Extensions;
+
+internal static class UserExtensions
+{
+    private static readonly IDictionary<string, string> EmptyDictionary =
+        new ReadOnlyDictionary<string, string>(new Dictionary<string, string>());
+
+    public static User ToUser(this SentryCocoa.SentryUser user)
+    {
+        IDictionary<string, string>? otherData;
+        if (user.Data is not IDictionary<NSString, NSObject> data)
+        {
+            otherData = EmptyDictionary;
+        }
+        else
+        {
+            otherData = data
+                .ToDictionary(
+                    x => (string)x.Key,
+                    x =>
+                    {
+                        try
+                        {
+                            return JsonSerializer.Serialize(x.Value);
```

Comment:
How successful is this in general? I have no experience with getting an NSObject serialized through S.T.J

In other SDKs we opt'ed to omit synchronizing this property due to risk/complexity.

Reply:
Good callout.  I'll investigate.

Reply:
Indeed, there is a better way to handle this.  I'll do that in a separate PR shortly.  Thanks.